src/Eloquent/Model.php

```diff
@@ -704,7 +708,7 @@ protected function addCastAttributesToArray(array $attributes, array $mutatedAtt
             }
 
             if ($this->isEnumCastable($key) && (! $castValue instanceof Arrayable)) {
-                $castValue = $castValue !== null ? $this->getStorableEnumValue($castValue) : null;
```

Comment:
The signature of this method have changed in a non-backward compatible way. So I duplicated this simple method to keep compatibility with both Laravel 10 and 11.
https://github.com/laravel/framework/commit/8647dcf18e6e315e97daea9ed60ef79b420ad327#diff-59d24f1a8f0dd1e51ee7dffc8778133711634e928ecef4a91b63fc3851cb1579R1188

We must keep the function `addCastAttributesToArray` introduced for Laravel 10 compatibility, as it provides casting of nested fields. https://github.com/mongodb/laravel-mongodb/commit/2ea1a7c0aee085c382349613d399be79a0bd1f1a#diff-284e164f76aeb13f424d154f9208d91fdbb0112b0a50ad8b1dfa237176f03cf2R525-R530